pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -303,7 +304,7 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 			miss += len(cached)
 
 			// We have a cache miss, so we will re-generate the cluster and later store it in the cache.
-			lbEndpoints := cb.buildLocalityLbEndpoints(clusterKey.proxyView, service, port.Port, nil)
+			lbEndpoints := clusterKey.endpointBuilder.FromServiceEndpoints()
```

Comment:
```suggestion
			var lbEndpoints []*endpoint.LocalityLbEndpoints
			// We have a cache miss, so we will re-generate the cluster and later store it in the cache.
			if (service.Resolution == model.DNSLB || service.Resolution == model.DNSRoundRobinLB) {
				lbEndpoints = clusterKey.endpointBuilder.FromServiceEndpoints()
			}
```

I think this fixes the performance regression, which is due to always computing endpoints even if its an EDS cluster.

Reply:
Right.. I did this on the original draft and un-did it. Thanks!

Reply:
Still +5% mean. The ones that have a higher regression are on the order of `us`.

Reply:
Turns out it was passing around/copying the `EndpointBuilder` struct. Made that a pointer and everything looks good.

Reply:
Everything is equal now.